### PR TITLE
[qontract-reconcile-base] upgrade terraform to v1

### PR DESCRIPTION
**Pull request by maorfr**
related to:
- https://issues.redhat.com/browse/APPSRE-9906
- https://issues.redhat.com/browse/APPSRE-8452

https://github.com/hashicorp/terraform/releases/tag/v1.5.7

### CODE REVIEW

File: qontract-reconcile-base/Dockerfile [hunk 1]
```diff
@@ -5,15 +5,15 @@ RUN dnf -y install unzip glibc-langpack-en
 ENV LC_ALL=en_US.utf8
 ENV LANG=en_US.utf8
 
-ENV TF_PROVIDER_AWS_VERSIONS="3.76.0 3.76.1 4.67.0"
-ENV TF_PROVIDER_CLOUDFLARE_VERSIONS="3.32.0"
+ENV TF_PROVIDER_AWS_VERSIONS="3.76.0 3.76.1 4.67.0 5.39.1"
+ENV TF_PROVIDER_CLOUDFLARE_VERSIONS="3.32.0 4.26.0"
```
Comment (rporres) [inline qontract-reconcile-base/Dockerfile:9]: why do we need these new versions? The migration is about terraform binary, the provider is a very different story AFAIK

Comment (maorfr) [reply qontract-reconcile-base/Dockerfile:9]: just to prepare things and avoid another PR to the base.

for visiblity, we want to upgrade cloudflare provider asap, but can take longer with aws.

Comment (rporres) [reply qontract-reconcile-base/Dockerfile:9]: I'd create a new MR whenever we're ready so things are not confused. This PR talks about terraform v1 migration, not provider migration. I don't know about cloudfare, but AWS one is not going to be trivial (if we can do it). And most likely we would move first to 4.x which is already included. Not a biggie, I just don't see it.

**Review by rporres — COMMENTED** [review]
Apart from the comment above, I assume you have been testing the v1 so this LGTM.

I'd focus the PR in the v1 thing to avoid confusion and do atomic changes

**Review by rporres — APPROVED** [review]

**Pull request merged**